plugins/out_kafka/kafka.c

```diff
@@ -151,6 +156,35 @@ int produce_message(struct flb_time *tm, msgpack_object *map,
                 topic = flb_kafka_topic_lookup((char *) val.via.str.ptr,
                                                val.via.str.size,
                                                ctx);
+                /* Add topic on the fly to topiclist and use it */
+                if (ctx->dynamic_topic) {
+                    if (strncmp(topic->name, flb_kafka_topic_default(ctx)->name, val.via.str.size) == 0 &&
+                       (strncmp(topic->name, val.via.str.ptr, val.via.str.size) != 0) ) {
+                        dynamic_topic = flb_malloc(val.via.str.size + 1);
+                        if (!dynamic_topic) {
+                            flb_errno();
+                            /* in this case we use the default topic */
+                            break;
+                        }
+                        strncpy(dynamic_topic, val.via.str.ptr, val.via.str.size);
+                        dynamic_topic[val.via.str.size] = '\0';
+
+                        topics = flb_utils_split(dynamic_topic, ',', 1);
```

Comment:
please check that ```topics``` is not NULL, otherwise raise the error and return

Reply:
NULL check added